lib/oban.ex

```diff
@@ -353,6 +381,38 @@ defmodule Oban do
   """
   defdelegate config, to: Config, as: :get
 
+  @doc """
+  Synchronously execute all available jobs in a queue. All execution happens within the current
+  process and it is guaranteed not to raise an error or exit.
+
+  Draining a queue from within the current process is especially useful for testing. Jobs that are
+  enqueued by a process when Ecto is in sandbox mode are only visible to that process. Calling
+  `drain_queue/1` allows you to control when the jobs are executed and to wait synchronously for
+  all jobs to complete.
+
+  ## Failures & Retries
+
+  Draining a queue uses the same execution mechanism as regular job dispatch. That means that any
+  job failures or crashes will be captured and result in a retry. Failures are _not_ reported back
```

Comment:
Are they retried immediately? Or will they be requeued and then I need to drain the queue again?

Reply:
There is a backoff for all jobs which causes them to be scheduled in the future. Draining only executes currently available jobs so they won't be retried immediately. I'll make that clear in the docs. 👍

Reply:
Explicitly needing to redrain the queue actually feels more explicit and like more control over testing stuff, which includes failures.

Reply:
@LostKobrakai Agreed, preventing an automatic retry seems more explicit to me. I'm having a hard time imagining a situation where, during an integration level test, a job would error and I would want to retry it again.